File: valid parenthesis.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        pairs = {')' : '(', '}' : '{', ']' : '['}
        inputs = ['(', '{', '[']
        stack = []
        for char in s:
            if char in inputs:
                stack.append(char)
            else:
                check = pairs[char]
    
                if len(stack) == 0 or check != stack[-1]:
                    return False
                else:
                    stack.pop()
        
        if len(stack) == 0:        
            return True
        return False
if __name__ == '__main__':
    s = Solution()
    print (s.isValid( "()"))

File: Program's_Contributed_By_Contributors/C++ Programs/Stacks&Queues/ValidParenthesis.cpp
//Leetcode 20.

class Solution {
public:
    bool isValid(string str) {
        stack <char> s;
        int len = str.length();

        for(int i=0; i<len; i++){
            if(str[i] == '(' || str[i] =='{' || str[i] == '[')
                s.push(str[i]);
            else if(str[i] == ')' || str[i] == '}' ||str[i] == ']'){
                if(s.empty())
                    return false;
                else if(str[i] == ')' && s.top() != '(')
                    return false;
                else if(str[i] == '}' && s.top() != '{')
                    return false;
                else if(str[i] == ']' && s.top() != '[')
                    return false;
                else 
                    s.pop();
            }
            
        }
        if(s.empty())
            return true;
        else 
            return false;
        
    }
};